opencl-aot/source/utils.cpp

```diff
@@ -240,6 +246,7 @@ getOpenCLPlatform(DeviceType Type) {
          DeviceTypesToSupportedPlatformNames[Type]) {
       ErrorMessage += "  " + SupportedPlatformName + '\n';
     }
+    CLErr = OPENCL_AOT_DEVICE_ID_IS_EMPTY;
```

Comment:
Are you sure that this is correct? I feel like this should be set only if we haven't found any platform, i.e. if `!ErrorMessage.empty()` by some reason.

One more minor comment: error code doesn't look related to the problem to me - I suggest to add one more error code for this particular situation

Reply:
> Are you sure that this is correct? I feel like this should be set only if we haven't found any platform, i.e. if `!ErrorMessage.empty()` by some reason.

This is correct. Lines 243-250:
```cpp
if (!ErrorMessage.empty()) {
  ErrorMessage += "Failed to find any of these OpenCL platforms:\n";
  for (const auto &SupportedPlatformName :
       DeviceTypesToSupportedPlatformNames[Type]) {
    ErrorMessage += "  " + SupportedPlatformName + '\n';
  }
  CLErr = OPENCL_AOT_DEVICE_ID_IS_EMPTY;
}
```

> One more minor comment: error code doesn't look related to the problem to me - I suggest to add one more error code for this particular situation

How does `OPENCL_AOT_PLATFORM_NOT_FOUND` sound?

Reply:
```OPENCL_AOT_PLATFORM_NOT_FOUND``` LGTM, you can add it to ```CLErr``` in case if ```ErrorMessage``` is not empty and ```CLErr``` is still ```CL_SUCCESS```

Reply:
> This is correct. Lines 243-250:

Sorry, I missed `{}` after `if (!ErrorMessage.empty())`. Indeed this is correct.

> How does OPENCL_AOT_PLATFORM_NOT_FOUND sound?

Looks good to me as well